main.tf

```diff
@@ -62,8 +81,10 @@ data "template_file" "aws_logs_policy" {
 resource "aws_s3_bucket" "aws_logs" {
   bucket = "${var.s3_bucket_name}"
 
-  acl    = "log-delivery-write"
+  acl    = "${var.enable_all_services || var.enable_s3  ? "log-delivery-write" : "private"}"
   region = "${var.region}"
+
+  # TODO - This needs to be unset for s3 log bucket - currently breaks
```

Comment:
I tried setting policy to "null" if enable_s3 was true, and passing an empty statement ({}) into the policy for creating an s3 logging bucket. Both did not work. We can add a block for enabling s3 policy to the bit that was removed back into the policy, but that's not ideal, given that we've determined that that policy isn't the right way to implement privacy. Not sure how to handle this use case.

Reply:
Solution was to create an `aws_s3_bucket_policy` and attach it to the bucket depending on the use case, i.e. don't attach a policy for an s3 bucket for s3 logging, but do for every other case.